src/dataflow/processor.ts

```diff
@@ -13,20 +13,23 @@ export interface DataflowProcessorInformation<OtherInfo> {
 	/**
    * Initial and frozen ast-information
    */
-	readonly completeAst:  NormalizedAst<OtherInfo>
+	readonly completeAst:      NormalizedAst<OtherInfo>
 	/**
    * Correctly contains pushed local scopes introduced by `function` scopes.
    * Will by default *not* contain any symbol-bindings introduces along the way, they have to be decorated when moving up the tree.
    */
-	readonly environments: REnvironmentInformation
+	readonly environments:     REnvironmentInformation
 	/**
    * Name of the currently active scope, (hopefully) always {@link LocalScope | Local}
    */
-	readonly activeScope:  DataflowScopeName
+	readonly activeScope:      DataflowScopeName
 	/**
    * Other processors to be called by the given functions
    */
-	readonly processors:   DataflowProcessors<OtherInfo>
+	readonly processors:       DataflowProcessors<OtherInfo>
+	// TODO using "initial" as the default path doesn't allow us to skip re-sourcing the initial file - how do we find out the initial file's name/path?
+	readonly currentPath:      string | 'initial'
```

Comment:
I'm really not sure if this is a good and/or valid way of dealing with "what file are we currently analyzing", since I also don't know if this is already being stored some way somewhere. Is this acceptable as a way of tracking this @EagleoutIce?

Reply:
It is how i would have done it as, currently, there is only the current 'request'.... Yet instead of a path we should probably store the retrieval request!

Reply:
That's a good idea, I was actually wondering what to do when the request is just text instead of a file, but just storing the request directly would solve that as well. Yay!